utils/lsblk.go

```diff
@@ -0,0 +1,107 @@
+package utils
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/bmc-toolbox/common"
+	"github.com/pkg/errors"
+)
+
+const lsblk = "/usr/bin/lsblk"
+
+var ErrLsblkTransportUnsupported = errors.New("Unsupported transport type")
+
+type Lsblk struct {
+	Executor Executor
+}
+
+type lsblkDeviceAttributes struct {
+	Name      string `json:"name"`
+	Device    string `json:"path"`
+	Model     string `json:"model"`
+	Serial    string `json:"serial"`
+	Firmware  string `json:"rev:"`
+	Transport string `json:"tran"`
+}
+
+type lsblkList struct {
+	Devices []*lsblkDeviceAttributes `json:"Devices"`
+}
+
+// Return a new lsblk executor
+func NewLsblkCmd(trace bool) *Lsblk {
+	e := NewExecutor(lsblk)
+	e.SetEnv([]string{"LC_ALL=C.UTF-8"})
+
+	if !trace {
+		e.SetQuiet()
+	}
+
+	return &Lsblk{Executor: e}
+}
+
+// Executes lsblk list, parses the output and returns a slice of *common.Drive
+func (l *Lsblk) Drives(ctx context.Context) ([]*common.Drive, error) {
+	drives := make([]*common.Drive, 0)
+
+	out, err := l.List()
+	if err != nil {
+		return nil, err
+	}
+
+	list := &lsblkList{Devices: []*lsblkDeviceAttributes{}}
+
+	err = json.Unmarshal(out, list)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, d := range list.Devices {
+		dModel := d.Model
+
+		var vendor string
+		var product []string
+
+		modelTokens := strings.Split(d.Model, " ")
+
+		if len(modelTokens) > 1 {
+			vendor = modelTokens[1]
+			product = append(modelTokens[:1], modelTokens[2:]...)
```

Comment:
The usage of `append()` here is incorrect, I'm not completely sure whats going on here here - but you most likely want to call append twice.

```
product = append(product, modelTokens[:1])
product = append(product, modelTokens[2:]...) 
```

Would be worth leaving a comment as to what this block is doing.

Reply:
This was a poorly executed attempt to remove the first element (vendor) and use the rest as the product name.  Since lshw will be populating this, we decided lsblk doesn't need to.  This has been removed.